bakery/src/scripts/disassemble-book.py

```diff
@@ -76,7 +76,65 @@ def main():
                 link.attrib['href'] = f'./{id_with_context}.xhtml'
 
         with open(f"{out_dir / id_with_context}.xhtml", "wb") as out:
-            out.write(bytes(DocumentContentFormatter(doc)))
+            # Inject some styling and JS for QA
+            val = str(DocumentContentFormatter(doc))
+            val = val.replace('<body', u'''
+                <head>
+                    <style>
+                        /* Linking to a specific element should highlight the element */
+                        :target {
+                            background-color: #ffffcc;
+                            border: 1px dotted #000000;
+
+                            animation-name: cssAnimation;
+                            animation-duration: 10s;
+                            animation-timing-function: ease-out;
+                            animation-delay: 0s;
+                            animation-fill-mode: forwards;
+                        }
+                        @keyframes cssAnimation {
+                            to {
+                                background-color: initial;
+                                border: initial;
+                            }
+                        }
+
+                        /* Style footnotes so that they stand out */
+                        [role="doc-footnote"] { background-color: #ffcccc; border: 1px dashed #ff0000; }
+                        [role="doc-footnote"]:before { content: "FOOTNOTE " ; }
+                        
+                        /* Show a permalink when hovering over a heading or paragraph */
+                        *:not(:hover) > a.-permalinker { display: none; }
+                        * > a.-permalinker {
+                            margin-left: .1rem;
+                            font-weight: bold;
+                            text-decoration: none;
+                        }
+                    </style>
+                </head>
+                <body''')
+            val = val.replace('</body>', u'''
+                <script>//<![CDATA[
+                
+                    const pilcrow = '¶'
+
+                    function addPermalink(parent, id) {
+                        const link = window.document.createElement('a')
+                        link.classList.add('-permalinker')
+                        link.setAttribute('href', `#${id}`)
+                        link.textContent = pilcrow
+                        parent.appendChild(link)
+                    }
+
+                    const paragraphs = Array.from(document.querySelectorAll('p[id]'))
+                    paragraphs.forEach(p => addPermalink(p, p.getAttribute('id')) )
+
+                    const headings = Array.from(document.querySelectorAll('*[id] > h1, *[id] > h2, *[id] > h3, *[id] > h4, *[id] > h5, *[id] > h6'))
+                    headings.forEach(h => addPermalink(h, h.parentElement.getAttribute('id')) )
+
+                // ]]></script>
+                </body>''')
```

Comment:
The JSON is the supported format... These XHTML files are just for dev/QA. Maybe they should have an additional disclaimer at the top with a watermark as the page background so that no one accidentally uses these files?